app/services/forest_liana/permissions_checker.rb

```diff
@@ -1,100 +1,131 @@
+      # if !have_permissions_expired && is_allowed
+      return true unless have_permissions_expired? || !is_allowed
 
-    def get_permissions
-      @@permissions_per_rendering &&
-        @@permissions_per_rendering[@rendering_id] &&
-        @@permissions_per_rendering[@rendering_id]['data']
+      fetch_permissions
+      is_allowed
     end
 
-    def get_last_retrieve
-      @@permissions_per_rendering &&
-        @@permissions_per_rendering[@rendering_id] &&
-        @@permissions_per_rendering[@rendering_id]['last_retrieve']
-    end
+    private
 
-    def smart_action_allowed?(smart_actions_permissions) 
-      if !@smart_action_parameters||
-          !@smart_action_parameters[:user_id] ||
-          !@smart_action_parameters[:action_id] ||
-          !smart_actions_permissions ||
-          !smart_actions_permissions[@smart_action_parameters[:action_id]]
-        return false
+    def fetch_permissions
+      permissions = ForestLiana::PermissionsGetter::get_permissions_for_rendering(@rendering_id)
+      @@roles_acl_activated = permissions['meta']['rolesACLActivated']
+      permissions['last_fetch'] = Time.now
+      if @@roles_acl_activated
+        @@permissions_cached = permissions
+      else
+        permissions['data'] = ForestLiana::PermissionsFormatter.convert_to_new_format(permissions['data'])
+        @@permissions_cached[@rendering_id] = permissions
       end
-
-      @user_id = @smart_action_parameters[:user_id]
-      @action_id = @smart_action_parameters[:action_id]
-      @smart_action_permissions = smart_actions_permissions[@action_id]
-      @allowed = @smart_action_permissions['allowed']
-      @users = @smart_action_permissions['users']
-
-      return @allowed && (@users.nil?|| @users.include?(@user_id.to_i));
     end
 
-    def collection_list_allowed?(scope_permissions)
-      return ForestLiana::ScopeValidator.new(
-        scope_permissions['filter'],
-        scope_permissions['dynamicScopesValues']['users']
-      ).is_scope_in_request?(@collection_list_parameters)
-    end
+    def is_allowed
+      permissions = get_permissions_content
 
-    def is_allowed?
-      permissions = get_permissions
       if permissions && permissions[@collection_name] &&
         permissions[@collection_name]['collection']
         if @permission_name === 'actions'
           return smart_action_allowed?(permissions[@collection_name]['actions'])
         # NOTICE: Permissions[@collection_name]['scope'] will either contains conditions filter and
         #         dynamic user values definition, or null for collection that does not use scopes
-        elsif @permission_name === 'list' and permissions[@collection_name]['scope']
+        # TODO: Handle scopes
+        elsif @permission_name === 'browseEnabled' and permissions[@collection_name]['scope']
+          # TODO: handle this
           return collection_list_allowed?(permissions[@collection_name]['scope'])
```

Comment:
I told u the `collection_list_allowed` is supposed to check that you have the browseEnabled permissions too so why did you resolve the message but I still see this condition 🤔 ?

Reply:
Make sure to fix a comment before resolving it please, I can't just recheck every time I pass 😅

Reply:
Sorry, I must have resolved this one by mistake